craft_parts/executor/step_handler.py

```diff
@@ -45,21 +45,44 @@
 
 
 @dataclasses.dataclass(frozen=True)
-class StepContents:
+class StepPartitionContents:
     """Files and directories to be added to the step's state."""
 
     files: set[str] = dataclasses.field(default_factory=set)
     dirs: set[str] = dataclasses.field(default_factory=set)
 
 
 @dataclasses.dataclass(frozen=True)
-class StageContents(StepContents):
+class StagePartitionContents(StepPartitionContents):
```

Comment:
This extends StepPartitionContents adding backstage states to the regular stage contents. Backstage isn't a separate step to have its own instance of StepPartitionContents.